### PR TITLE
Provide RequestMessage parameter to allowlist handler functions

### PR DESCRIPTION
**Pull request by lgostling**
Provide RequestMessage parameter to allowlist handler function

### CODE REVIEW

File: Core.Collectors/Web/RequestResult.cs [hunk 1]
```diff
@@ -6,19 +6,20 @@ namespace Microsoft.CloudMine.Core.Collectors.Web
 {
     public class RequestResult
     {
+        public HttpRequestMessage request;
         public HttpResponseMessage response;
         public IAllowListStatus allowListStatus;
```
Comment (kivancmuslu) [inline Core.Collectors/Web/RequestResult.cs:11]: Any field that is not accessed outside, please make private or private read only.

File: Core.Collectors/Web/RequestResult.cs [hunk 2]
```diff
@@ -6,19 +6,20 @@ namespace Microsoft.CloudMine.Core.Collectors.Web
 {
     public class RequestResult
     {
+        public HttpRequestMessage request;
```
Comment (kivancmuslu) [inline Core.Collectors/Web/RequestResult.cs:9]: If this is accessed outside, this either needs to be property (with private set and public get) or there should be a method accessing it.

File: Core.Collectors/Web/RequestResult.cs [hunk 3]
```diff
@@ -6,19 +6,20 @@ namespace Microsoft.CloudMine.Core.Collectors.Web
...
         public HttpResponseMessage response;
         public IAllowListStatus allowListStatus;
 
-        public RequestResult(HttpResponseMessage response)
-                : this(response, null)
+        public RequestResult(HttpRequestMessage requestMessage, HttpResponseMessage response)
```
Comment (kivancmuslu) [inline Core.Collectors/Web/RequestResult.cs:13]: >HttpResponseMessage response [](http://example.com/codeflow?start=64&length=28)

Just wondering: doesn't response already have the request? If you can get the request (URL) from response, then you don't need to pass the request again, no?

File: Core.Collectors/Web/HttpExceptionSignature.cs [hunk 4]
```diff
@@ -4,21 +4,27 @@
...
 using System.Collections.Generic;
+using System.Net.Http;
 using System.Threading.Tasks;
 
 namespace Microsoft.CloudMine.Core.Collectors.Web
 {
     public class HttpExceptionSignature : IAllowListStatus
     {
-        public static readonly HttpExceptionSignature RequestTimeoutException = new HttpExceptionSignature(exception =>
+        public static HttpExceptionSignature RequestTimeoutException(Func<HttpRequestMessage, List<CollectionNode>> continuationNodeList = null)
```
Comment (kivancmuslu) [inline Core.Collectors/Web/HttpExceptionSignature.cs:14]: >Func<HttpRequestMessage, List<CollectionNode>> continuationNodeList = null [](http://example.com/codeflow?start=69&length=74)

As usual, can we leave this change (which is not referenced here) to a separate PR?)

File: Core.Collectors/Web/RequestResult.cs (hunk 3 above)
Comment (lgostling) [reply Core.Collectors/Web/RequestResult.cs:13]: The response does contain the request, however in the case of an allow-listed exception there will not be a response for us to use.

**Comment by kivancmuslu** [comment]
Please take a look and see if the RequestResponse (which we already have) can be re-used to get the request URL. If so, we don't need a separate field. Thanks!

**Review by kivancmuslu — APPROVED** [review]
:shipit:

File: Core.Collectors/Web/RequestResult.cs (hunk 3 above)
Comment (lgostling) [reply Core.Collectors/Web/RequestResult.cs]: I have changed this constructor to take only an HttpResponseMessage and then filled both request and response fields using that object.

**Pull request merged**